zsh-history-substring-search.zsh

```diff
@@ -247,11 +247,8 @@ _history-substring-search-begin() {
     # Escape and join query parts with wildcard character '*' as seperator
     # `(j:CHAR:)` join array to string with CHAR as seperator
     #
-    if [[ -n $HISTORY_SUBSTRING_SEARCH_PREFIX ]]; then
-        local search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
-    else
-        local search_pattern="*${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
-    fi
+    search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
```

Comment:
You still need to keep the `local` keyword, to avoid conflicts with global or user variables having the same name.

Reply:
Fixed.